internal/services/mobilenetwork/mobile_network_sim_group_resource.go

```diff
@@ -64,7 +64,7 @@ func (r SimGroupResource) Arguments() map[string]*pluginsdk.Schema {
 			ValidateFunc: validation.IsURLWithHTTPorHTTPS,
 		},
 
-		"identity": commonschema.SystemOrUserAssignedIdentityOptional(),
+		"identity": commonschema.UserAssignedIdentityOptional(),
```

Comment:
we also need to update the Expand and Flatten functions to account for this?

Reply:
the typed model is correct so I think these functions are ok?

Reply:
so if the service were to support system assigned, the provider would crash due to the flattened schema not matching what's defined in the schema - we'll need to ensure these are all using the `user assigned` values here

Reply:
I dont think it will crash? with the flatten code accept both SystemAssigned and UserAssigned but only use the UserAssgined one .
```
func flattenMobileNetworkUserAssignedToNetworkLegacyIdentity(input *identity.LegacySystemAndUserAssignedMap) ([]identity.ModelUserAssigned, error) {
	if input == nil {
		return nil, nil
	}
	tmp := identity.UserAssignedMap{
		Type:        input.Type,
		IdentityIds: input.IdentityIds,
	}
	output, err := identity.FlattenUserAssignedMapToModel(&tmp)
	if err != nil {
		return nil, fmt.Errorf("expanding `identity`: %+v", err)
	}
	return *output, nil
}
```